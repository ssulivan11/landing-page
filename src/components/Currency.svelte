<script>
  import { onMount } from 'svelte'
  import * as customJson from '../../custom.json'

  let dollarConversion
  if (customJson.currency) {
    onMount(async () => {
      const res = await fetch(
        `https://api.exchangeratesapi.io/latest?base=${customJson.currency.convertFrom}`
      )
      dollarConversion = await res.json()
      dollarConversion = dollarConversion.rates.USD.toFixed(3)

      if (
        customJson.currency.alertAt &&
        dollarConversion > customJson.currency.alertAt
      ) {
        alert('The Sterling Pound is over $1.30 conversion!')
      }
    })
  }
</script>

<style>
  .currency {
    text-decoration: none;
  }
  .currency:hover {
    text-decoration: underline;
  }
</style>

{#if dollarConversion}
  <a
    class="currency"
    href={`https://www.xe.com/currencyconverter/convert/?Amount=1&From=${customJson.currency.convertFrom}&To=USD`}
    target="_blank">
    {customJson.currency.convertFrom} ${dollarConversion}
  </a>
{/if}
