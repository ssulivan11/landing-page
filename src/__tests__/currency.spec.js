import { tick } from 'svelte'
import { render } from '@testing-library/svelte'
import promiseCurrencyMock from '../__mocks__/promise.mock'
import Currency from '../components/Currency.svelte'

test('should render Currency with mock fetch', async () => {
  promiseCurrencyMock(1.2)
  const { container } = render(Currency)
  await tick()
  await tick() // for onMount..
  expect(container.querySelector('.currency')).toMatchSnapshot()
})

test('should alert when currency exchange is over $1.5', async () => {
  promiseCurrencyMock(1.5)
  global.alert = jest.fn()
  render(Currency)
  await tick()
  await tick() // for onMount..
  expect(global.alert).toHaveBeenCalledTimes(1)
})
