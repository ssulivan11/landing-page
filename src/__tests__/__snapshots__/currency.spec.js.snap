// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Currency with mock fetch 1`] = `
<a
  class="currency svelte-o1cyf9"
  href="https://www.xe.com/currencyconverter/convert/?Amount=1&From=GBP&To=USD"
>
  GBP
   $
  1.200
</a>
`;
